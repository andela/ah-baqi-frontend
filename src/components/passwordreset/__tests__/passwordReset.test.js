import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import rootReducer from '../../../reducers/index';
import PasswordReset from '../passwordResetContainer';

const configureStoreItem = (initialState) => {
  const store = applyMiddleware(thunk)(createStore);
  return store(rootReducer, initialState);
};

describe('<Content />', () => {
  const store = configureStoreItem();
  test('It renders correctly', () => {
    const wrapperComponent = mount(
      <Provider store={store}>
        <BrowserRouter>
          <PasswordReset />
        </BrowserRouter>
      </Provider>,
    );
    const formItem = wrapperComponent.find('[data-test="reset-form-cont"]');
    expect(formItem).toHaveLength(2);

    const event = { target: { name: 'password', value: 'Whf34@jfFFdjkv34%#' }, preventDefault: jest.fn() };
    formItem.at(0).props().onSubmit(event);

    const resetPass = wrapperComponent.find('[data-test="email-reset-passwor"]');
    resetPass.at(0).props().onChange(event);
    expect(resetPass).toHaveLength(2);
  });
});
