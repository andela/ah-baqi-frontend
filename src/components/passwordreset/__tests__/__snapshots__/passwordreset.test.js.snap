// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests PasswordResetContainer Tests mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Form(PasswordReset))
      showModal={[MockFunction]}
    >
      <Form(PasswordReset)
        resetPasswordActions={[Function]}
        showModal={[MockFunction]}
      >
        <PasswordReset
          form={
            Object {
              "getFieldDecorator": [Function],
              "getFieldError": [Function],
              "getFieldInstance": [Function],
              "getFieldProps": [Function],
              "getFieldValue": [Function],
              "getFieldsError": [Function],
              "getFieldsValue": [Function],
              "isFieldTouched": [Function],
              "isFieldValidating": [Function],
              "isFieldsTouched": [Function],
              "isFieldsValidating": [Function],
              "isSubmitting": [Function],
              "resetFields": [Function],
              "setFields": [Function],
              "setFieldsInitialValue": [Function],
              "setFieldsValue": [Function],
              "submit": [Function],
              "validateFields": [Function],
              "validateFieldsAndScroll": [Function],
            }
          }
          resetPasswordActions={[Function]}
          showModal={[MockFunction]}
        >
          <Form
            className="reset-form"
            colon={true}
            hideRequiredMark={false}
            layout="horizontal"
            onSubmit={[Function]}
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <form
              className="ant-form ant-form-horizontal reset-form"
              onSubmit={[Function]}
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <div>
                <h1>
                  Forgot Password
                </h1>
              </div>
              <FormItem
                className="reset-label"
                hasFeedback={false}
                label="Email"
              >
                <Row
                  className="ant-form-item reset-label"
                  gutter={0}
                  key="row"
                >
                  <div
                    className="ant-row ant-form-item reset-label"
                  >
                    <Col
                      className="ant-form-item-label"
                    >
                      <div
                        className="ant-col ant-form-item-label"
                      >
                        <label
                          className="ant-form-item-required"
                          htmlFor="email"
                          onClick={[Function]}
                          title="Email"
                        >
                          Email
                        </label>
                      </div>
                    </Col>
                    <Col
                      className="ant-form-item-control-wrapper"
                    >
                      <div
                        className="ant-col ant-form-item-control-wrapper"
                      >
                        <div
                          className="ant-form-item-control"
                        >
                          <span
                            className="ant-form-item-children"
                          >
                            <Input
                              data-__field={
                                Object {
                                  "name": "email",
                                }
                              }
                              data-__meta={
                                Object {
                                  "name": "email",
                                  "originalProps": Object {
                                    "name": "email",
                                    "onChange": [Function],
                                    "placeholder": "Please enter your email ",
                                    "prefix": <Icon
                                      style={
                                        Object {
                                          "color": "rgba(0,0,0,.25)",
                                        }
                                      }
                                      type="mail"
                                    />,
                                    "required": true,
                                    "type": "text",
                                  },
                                  "ref": null,
                                  "rules": Array [
                                    Object {
                                      "message": "Please enter a valid email",
                                      "type": "email",
                                    },
                                    Object {
                                      "message": "Please input your email",
                                      "required": true,
                                    },
                                  ],
                                  "trigger": "onChange",
                                  "validate": Array [
                                    Object {
                                      "rules": Array [
                                        Object {
                                          "message": "Please enter a valid email",
                                          "type": "email",
                                        },
                                        Object {
                                          "message": "Please input your email",
                                          "required": true,
                                        },
                                      ],
                                      "trigger": Array [
                                        "onChange",
                                      ],
                                    },
                                  ],
                                  "valuePropName": "value",
                                }
                              }
                              id="email"
                              name="email"
                              onChange={[Function]}
                              placeholder="Please enter your email "
                              prefix={
                                <Icon
                                  style={
                                    Object {
                                      "color": "rgba(0,0,0,.25)",
                                    }
                                  }
                                  type="mail"
                                />
                              }
                              required={true}
                              type="text"
                            >
                              <span
                                className="ant-input-affix-wrapper"
                              >
                                <span
                                  className="ant-input-prefix"
                                >
                                  <Icon
                                    style={
                                      Object {
                                        "color": "rgba(0,0,0,.25)",
                                      }
                                    }
                                    type="mail"
                                  >
                                    <LocaleReceiver
                                      componentName="Icon"
                                    >
                                      <i
                                        aria-label="icon: mail"
                                        className="anticon anticon-mail"
                                        style={
                                          Object {
                                            "color": "rgba(0,0,0,.25)",
                                          }
                                        }
                                      >
                                        <IconReact
                                          className=""
                                          type="mail-o"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            className=""
                                            data-icon="mail"
                                            fill="currentColor"
                                            focusable="false"
                                            height="1em"
                                            key="svg-mail"
                                            viewBox="64 64 896 896"
                                            width="1em"
                                          >
                                            <path
                                              d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                                              key="svg-mail-svg-0"
                                            />
                                          </svg>
                                        </IconReact>
                                      </i>
                                    </LocaleReceiver>
                                  </Icon>
                                </span>
                                <input
                                  className="ant-input"
                                  data-__field={
                                    Object {
                                      "name": "email",
                                    }
                                  }
                                  data-__meta={
                                    Object {
                                      "name": "email",
                                      "originalProps": Object {
                                        "name": "email",
                                        "onChange": [Function],
                                        "placeholder": "Please enter your email ",
                                        "prefix": <Icon
                                          style={
                                            Object {
                                              "color": "rgba(0,0,0,.25)",
                                            }
                                          }
                                          type="mail"
                                        />,
                                        "required": true,
                                        "type": "text",
                                      },
                                      "ref": null,
                                      "rules": Array [
                                        Object {
                                          "message": "Please enter a valid email",
                                          "type": "email",
                                        },
                                        Object {
                                          "message": "Please input your email",
                                          "required": true,
                                        },
                                      ],
                                      "trigger": "onChange",
                                      "validate": Array [
                                        Object {
                                          "rules": Array [
                                            Object {
                                              "message": "Please enter a valid email",
                                              "type": "email",
                                            },
                                            Object {
                                              "message": "Please input your email",
                                              "required": true,
                                            },
                                          ],
                                          "trigger": Array [
                                            "onChange",
                                          ],
                                        },
                                      ],
                                      "valuePropName": "value",
                                    }
                                  }
                                  id="email"
                                  name="email"
                                  onChange={[Function]}
                                  onKeyDown={[Function]}
                                  placeholder="Please enter your email "
                                  required={true}
                                  style={null}
                                  type="text"
                                  value=""
                                />
                              </span>
                            </Input>
                          </span>
                          <Animate
                            animation={Object {}}
                            component=""
                            componentProps={Object {}}
                            key="help"
                            onAppear={[Function]}
                            onEnd={[Function]}
                            onEnter={[Function]}
                            onLeave={[Function]}
                            transitionAppear={true}
                            transitionEnter={true}
                            transitionLeave={true}
                            transitionName="show-help"
                          />
                        </div>
                      </div>
                    </Col>
                  </div>
                </Row>
              </FormItem>
              <div>
                <Button
                  block={true}
                  className="login-form-button"
                  ghost={false}
                  htmlType="submit"
                  loading={false}
                  type="primary"
                >
                  <Wave>
                    <button
                      className="ant-btn login-form-button ant-btn-primary ant-btn-block"
                      onClick={[Function]}
                      type="submit"
                    >
                      <span
                        key=".0"
                      >
                        Find Account
                      </span>
                    </button>
                  </Wave>
                </Button>
                <br />
                <br />
                <Link
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <Button
                      block={true}
                      className="cancel-reset-button login-form-button"
                      ghost={false}
                      htmlType="button"
                      loading={false}
                      type="danger"
                    >
                      <Wave>
                        <button
                          className="ant-btn cancel-reset-button login-form-button ant-btn-danger ant-btn-block"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            key=".0"
                          >
                            Cancel
                          </span>
                        </button>
                      </Wave>
                    </Button>
                  </a>
                </Link>
              </div>
            </form>
          </Form>
        </PasswordReset>
      </Form(PasswordReset)>
    </Connect(Form(PasswordReset))>
  </Provider>
</Router>
`;
